### ISSUE TITLE
Add special handling for [Intrinsics] methods

### ISSUE THREAD

**Issue by marek-safar** [mono/linker#762]
Add opt-in optimization to remove bodies of methods which are marked as [Intrinsics](https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/shared/System/Runtime/CompilerServices/IntrinsicAttribute.cs) which is internal CoreLib attribute. They are usually not big but their count is growing constantly.

**Comment by MichalStrehovsky** [comment]
`[Intrinsic]` doesn't mean the body is unused. If the body is guaranteed unused, we can just mark the method `extern` in C# source - no need for a linker step.

E.g. removing body for `Activator.CreateInsteance<T>` (that is marked Intrinsic) would make the method stop working for reference types and valuetypes with a default constructor. Intrinsic often only means we do something special when optimizing, etc.

`[Intrinsic]` really only says "the name of this method/type is special, you should look at it". Most places (runtime & compiler) cache the `IsIntrinsic` bit: marking things intrinsics avoids having to do expensive name matches.

**Comment by marek-safar** [comment]
> If the body is guaranteed unused, we can just mark the method extern in C# source - no need for a linker step.

I agree this would be a better approach but it's not the case today. Only handfull of [Intrinsic] are partially rewritten but the majority is fully rewritten.

**Comment by MichalStrehovsky** [comment]
> Only handfull of [Intrinsic] are partially rewritten but the majority is fully rewritten.

How will IL Linker know whether the body is safe to remove?
